### PR TITLE
Switch from Azure Pipelines to GitHub Actions

### PR DESCRIPTION
**Pull request by nifoc**
- Converted all steps from Azure Pipelines to GitHub Actions
- Added support for running tests using different Erlang versions

    Linux: 21, 22, 23
    macOS: 21, 22, latest (Hombrew)
    Windows: latest (Chocolatey)

- Remove README badge

### CODE REVIEW

**Comment by nifoc** [comment]
This can be seen in action on my fork of the project: [nifoc#re Actions](https://github.com/nifoc/re2/actions)

File: .github/workflows/re2.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - run: brew install ${{matrix.otp}} rebar3
+      - name: Compile
+        run: rebar3 compile
+      - name: Run tests
+        run: rebar3 eunit
+
+  test-windows:
+    runs-on: windows-2016
```
Comment (tuncer) [inline .github/workflows/re2.yml:70]: Would it make sense to use windows-latest here?

**Comment by tuncer** [comment]
To follow convention, maybe rename re2.yml to test.yml?

**Comment by nifoc** [comment]
Regarding `windows-latest`:

`windows-latest` is Windows Server 2019 and `re2` currently doesn't compile on that version (because of some CMake issue that I haven't looked into deeper).

See: https://github.com/nifoc/re2/runs/727138936 (Compile step)

**Comment by tuncer** [comment]
Looks like the cmake generator selected in `c_src/build_deps.sh` needs to be adapted. Feel free to open a ticket or post a diff.

**Pull request merged**